dev_guide/builds/buildconfig_environment.adoc

```diff
@@ -0,0 +1,147 @@
+[[dev-guide-build-buildconfig-environment]]
+= BuildConfig Environment
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[NOTE]
+====
+You can also manage environment variables defined in the `BuildConfig` with the
+xref:../../dev_guide/environment_variables.adoc#dev-guide-environment-variables[`oc set env`] command.
+====
+
+[[setting-environment-variables]]
+== Setting Environment Variables
+
+[source,yaml]
+----
+env:
+  - name: BUILD_LOGLEVEL <1>
+    value: "5" <2>
+----
+
+<1> The name to assign to the environment variable
+<2> The value to assign to the environment variable
+
+
+[[referencing-other-environment-variables]]
+== Referencing Other Environment Variables
+
+You can reference the value of a previously defined environment
+variable by using the `$()` syntax. If the environment variable
+reference can not be resolved, the value will be set to the provided
+string as shown below.
```

Comment:
it doesn't seem like we should have to be documenting this.  is this behavior (which is general k8s env var behavior) not already covered elsewhere in the docs?  if not, i would propose that it should be and we shouldn't have to say anything about it specific to builds.